tracer/src/Datadog.Trace/ClrProfiler/CallTarget/Handlers/IntegrationMapper.cs

```diff
@@ -75,7 +73,7 @@ internal static DynamicMethod CreateBeginMethodDelegate(Type integrationType, Ty
 
             List<Type> callGenericTypes = new List<Type>();
 
-            bool mustLoadInstance = onMethodBeginParameters.Length != argumentsTypes.Length;
+            bool mustLoadInstance = onMethodBeginParameters[0].ParameterType == genericArgumentsTypes[0];
```

Comment:
I think `onMethodBeginParameters.Length` can be null if we were instrumenting a static method with no parameters 🤔 I'm also not _entirely_ convinced by the check here, as it assumes that we have used the correct convention of the first generic argument being the type. I guess there's also the outside chance that a _static_ method, takes an instance of the type e.g.

```
public class MyType
{
    public static void TestFunc(MyType type) { /**/ }
}
```

Can't think of a _better_ check right now though  😅

Reply:
Ah thanks for catching that
I never thought about a `static` method taking an instance type of the same class.

I'll look into this today some more to see if I can come up with a better check